### PR TITLE
Simplify client code (tests) for the WinrtServer using winrt projections

### LINKED ISSUE #28: WinrtServer winrt C++ projections are not available in client projects
The TutorialsAndTests project only seem to access WinrtServer through the low-level `ABI` interface, instead of the "modern" `winrt` C++ projection that automatically converts HRESULTs to exceptions and output arguments to return-values. This makes the client code more complex than neccesary.

Root-cause seem to be the `winmdidl` call in CreateCppInterop.bat. I think replacing it with a `cppwinrt.exe` call (found in Microsoft.Windows.CppWinRT NuGet) would make the generated headers include `winrt` C++ projections. Alternatively, just copy the already generated headers from the `WinrtServer\Generated Files\winrt` folder.

### PR DESCRIPTION
**Pull request by jogerh**
Using 'modern' C++ winrt projections to simplify WinrtServer client code. Fixes #28

### CODE REVIEW

**Comment by jogerh** [comment]
@forderud this is an attempt to use the generated winrt headers to simplify client code for the WinrtServer. Currently the necessary header files are cherry-picked from the folder containing the full winrt library.

**Review by forderud — APPROVED** [review]
Very nice. The extra xcopy command just for WinrtServer.h seem a bit ad-hoc, but I guess it's still acceptable.

File: WinrtServer/CreateCppInterop.bat [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
+if exist %OutDir%Include\WinrtServer\winrt\ (
+    del /s /q %OutDir%Include\WinrtServer\winrt\ || exit /b 1
 )
-winmdidl %OutDir%\WinrtServer.winmd /outdir:%OutDir%Include\WinrtServer\ || exit \b 1
-pushd %OutDir%Include\WinrtServer || exit /b 1
-midlrt WinrtServer.idl /metadata_dir C:\Windows\System32\WinMetadata /ns_prefix || exit /b 1
\ No newline at end of file
+
+REM Copy the primary WinrtServer.h. This is a workaround for include path issues in generated files
+xcopy /I "Generated Files\winrt\WinrtServer.h" %OutDir%Include\WinrtServer\
```
Comment (forderud) [inline WinrtServer/CreateCppInterop.bat:8]: I think you can get rid of this extra xcopy if you change the client code includes to `#include <winrt/WinrtServer.h> (or similar)`

**Comment by jogerh** [comment]
Yes, it is not ideal, but I can look more at this some other time. The problem is in WinrtServer.h 

    #include "winrt/impl/WinrtServer.2.h"

So it has to be located in a folder above the 'winrt' folder. I'll look more into configuration possibilities later.

**Comment by forderud** [comment]
Ok. 👍

**Pull request merged**